### PR TITLE
Remove four old path functions

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/CmdlineSprite.cpp [hunk 1]
```diff
@@ -630,7 +630,7 @@ int32_t cmdline_for_sprite(const char** argv, int32_t argc)
                                                                               : ImageImporter::Palette::OpenRCT2;
             bool forceBmp = !jsonSprite["palette"].is_null() && Json::GetBoolean(jsonSprite["forceBmp"]);
 
-            auto imagePath = Path::GetAbsolute(std::string(directoryPath) + "/" + strPath);
+            auto imagePath = Path::GetAbsolute(directoryPath + u8"/" + strPath);
```
Comment (duncanspumpkin) [inline src/openrct2/CmdlineSprite.cpp:633]: This doesn't look safe should be using Path::Combine i think.

File: src/openrct2/core/Path.cpp [hunk 2]
```diff
@@ -80,6 +80,11 @@ namespace Path
         return u8path(path).extension().u8string();
     }
 
+    u8string WithExtension(u8string_view path, u8string_view newExtension)
+    {
+        return u8path(path).replace_extension(u8path(newExtension)).u8string();
```
[the remark this answers is not in the document]
Comment (duncanspumpkin) [reply src/openrct2/core/Path.cpp:85]: https://en.cppreference.com/w/cpp/filesystem/path/replace_extension

It appends the extension if it didn't have one before.

**Review by duncanspumpkin — APPROVED** [review]
Looks fine to me. Address the combine in a follow on if you want.

**Comment by IntelOrca** [comment]
Removed my review as @duncanspumpkin has already reviewed it.

**Pull request merged**